packages/flutter_tools/lib/src/build_system/targets/web.dart

```diff
@@ -241,13 +241,12 @@ class Dart2WasmTarget extends Dart2WebTarget {
     }
     final WasmCompilerConfig compilerConfig = WasmCompilerConfig.fromBuildSystemEnvironment(environment.defines);
     final BuildMode buildMode = BuildMode.fromCliName(buildModeEnvironment);
-    final Artifacts artifacts = globals.artifacts!;
+    final Artifacts artifacts = environment.artifacts;
     final File outputWasmFile = environment.buildDir.childFile(
       compilerConfig.runWasmOpt ? 'main.dart.unopt.wasm' : 'main.dart.wasm'
     );
     final File depFile = environment.buildDir.childFile('dart2wasm.d');
     final String dartSdkPath = artifacts.getArtifactPath(Artifact.engineDartSdkPath, platform: TargetPlatform.web_javascript);
-    final String dartSdkRoot = environment.fileSystem.directory(dartSdkPath).parent.path;
```

Comment:
This is...weird. It's going to the parent of the Dart sdk path...which is just the `bin/cache` dir. Replacing this below with `dartSdkPath` seems to work fine.

Reply:
Huh... this might be a holdover from before when the build directory structure was slightly different. The main thing this should affect is whether `--local-web-sdk` works when compiling to wasm... Which we both know doesn't work unless you do a bunch of weird symlinking. Would you mind trying doing a wasm build with `--local-web-sdk` without the `dart-sdk` symlink in the `wasm_release` folder? I'm curious whether your change actually resolves that problem.

Reply:
Yeah I'll give it a shot later today